### PR TITLE
Update auth and user profile for v1 API

### PR DESCRIPTION
**Pull request by cheungpat**

### CODE REVIEW

File: auth/basics/android.md [hunk 1]
```diff
@@ -227,40 +258,97 @@ skygear.getAuth().whoami(new AuthResponseHandler() {
 
 ## Updating the username / email of a user
 
-To change a user's username and email, you can use the `skygear.saveUser()` method by providing the user ID and the new username and/or the new email.
+## Updating a user's username and email
+
```
Comment (tensiuyan) [inline auth/basics/android.md]: Duplicated title.

File: auth/basics/ios.md [hunk 2]
```diff
@@ -144,110 +221,277 @@ Of course, you also would want to allow users to log out of their accounts by us
...
+    print("password changed successfully")
 }
-operation?.container = SKYContainer.default()
-SKYContainer.default().add(operation)
 ```
+
+Note: Changing the password of the current user will not cause
+[`SKYContainerDidChangeCurrentUserNotification`] notification to be posted.
+
+## Resetting password
```
Comment (tensiuyan) [inline auth/basics/ios.md]: Is resetting password related to forgot password?

**Comment by tensiuyan** [comment]
Thanks Kuen Kuen for the work.

I've compared the 3 user auth basics, and found that in the Android guide, "Observing user changes" is missing. May I just ask if Android doesn't have this API?

Thanks!

File: auth/basics/ios.md (hunk 2 above)
Comment (Steven-Chan) [reply auth/basics/ios.md]: @tensiuyan Yes, they work together as one feature. Forgot password provides the data you can call with reset password, and after calling reset password, the user would be able to login again.

Comment (Steven-Chan) [reply auth/basics/ios.md]: I think the point is `Not yet implemented.`

Comment (tensiuyan) [reply auth/basics/ios.md]: Thanks Steven. I think so. If that's the case shall we fill in this part in this pull request also? As the forgot password feature is already implemented.

**Comment by cheungpat** [comment]
@tensiuyan I don’t think the Android has support for observing user changes as far as I know.

File: auth/basics/ios.md (hunk 2 above)
Comment (cheungpat) [reply auth/basics/ios.md]: I could change it to “Coming soon” and we can write that part in a separate pull request. I can’t promise I could have this part ready before our release today.

**Pull request merged**